packages/vscode-extension/src/handlers.ts

```diff
@@ -612,27 +627,90 @@ export async function treeViewLocalDebugHandler(args?: any[]): Promise<Result<nu
 
 export async function treeViewPreviewHandler(env: string): Promise<Result<null, FxError>> {
   ExtTelemetry.sendTelemetryEvent(TelemetryEvent.TreeViewPreviewStart);
-  const LocalEnvName = environmentManager.getLocalEnvName();
-  const PreviewLocalSteps = 2;
-  const PreviewRemoteSteps = 1;
 
-  const progressBar = VS_CODE_UI.createProgressBar(
-    localize("teamstoolkit.preview.progressTitle"),
-    env === LocalEnvName ? PreviewLocalSteps : PreviewRemoteSteps
-  );
-  await progressBar.start();
+  let progressBar: IProgressHandler | undefined = undefined;
+  try {
+    const platformSingleSelect: SingleSelectQuestion = {
+      type: "singleSelect",
+      name: "hub",
+      title: localize("teamstoolkit.preview.platform.title"),
+      staticOptions: [constants.Hub.teams, constants.Hub.outlook, constants.Hub.office],
+      placeholder: localize("teamstoolkit.preview.platform.placeholder"),
+    };
+    const node = new QTreeNode(platformSingleSelect);
+    const inputs = getSystemInputs();
+    node.addChild(selectTeamsAppManifestQuestion(inputs));
+    const res = await traverse(node, inputs, VS_CODE_UI, ExtTelemetry.reporter);
```

Comment:
Does this have concurrent lock?
E.g., does remote preview work when there's a running provision or deploy progress?

Reply:
Good question. It does not have concurrent lock since all things are done in extension part (i.e., does not use `ConcurrentLockerMW`).
Not sure if there is a need to add concurrent lock because preview only reads env and does not write anything, like preview from F5.

Reply:
I would prefer not to add concurrent lock since it's a read-only operation.